pkg/network/node/networkpolicy.go

```diff
@@ -221,14 +231,63 @@ func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
 				for _, ip := range npp.selectedIPs {
 					if !selectedIPs.Has(ip) {
 						selectedIPs.Insert(ip)
-						otx.AddFlow("table=80, priority=100, reg1=%d, ip, nw_dst=%s, actions=drop", npns.vnid, ip)
+						allFlows = append(allFlows, fmt.Sprintf("table=80, priority=100, reg1=%d, ip, nw_dst=%s, actions=drop", npns.vnid, ip))
 					}
 				}
 			}
 
-			otx.AddFlow("table=80, priority=50, reg1=%d, actions=output:NXM_NX_REG2[]", npns.vnid)
+			allFlows = append(allFlows, fmt.Sprintf("table=80, priority=50, reg1=%d, actions=output:NXM_NX_REG2[]", npns.vnid))
 		}
 	}
+	
+	// calculate new flows
+	flowDiff := difference(allFlows, npns.flows)
+	for _, flow := range flowDiff {
+		otx.AddFlow(flow)
+	}
+
+	glog.V(5).Infof("Flow count npns.flows: %d allFlows: %d flowDiff: %d", len(npns.flows), len(allFlows), len(flowDiff))
+
+	// set the flows
+	npns.flows = allFlows
+	
+	// check current flows for ips that are not available
+	currentFlows, err := np.node.oc.ovs.DumpFlows("table=80")
```

Comment:
Since you computed the full set of flows that we currently want above (`allFlows`), and we have the set of currently-existing flows (`npns.flows`), then the set of flows that we want to delete should be `difference(npns.flows, allFlows)`, and you don't need to look at `DumpFlows` output, right?

(Also, the code here only removes `podSelector`-based flows, but leaves out-of-date `namespaceSelector`-based flows in place still, whereas diffing against the old flows ought to handle both.)

Reply:
The idea behind this cleanup was to call DumpFlows as our single source of truth and to make sure there is no garbage left in the OVS.
I was primary focused on the podSelector but yes I left out the namespaceSelector so we have to find a way to cleanup namespace related rules.